File: Euclid/Euclidean.cpp
#include<iostream>
using namespace std;

int Euclidean(int a, int b);
int main()
{
    int a, b;
    cout<<"Enter two numbers: "<<endl;
    cout<<"a = ";
    cin>>a;
    cout<<"b = ";
    cin>>b;
    cout<<"GCD of "<<a<<" and "<<b<<" is "<<Euclidean(a, b)<<endl;
}
//Recursive function to find the GCD of two numbers
int EuclideanRecursion(int a, int b)
{
    if(b==0) return a;
    return Euclidean(b, a%b);
}
//Iterative function to find the GCD of two numbers
int Euclidean(int a, int b)
{
    while(b!=0)
    {
        int temp = b;
        b = a%b;
        a = temp;
    }
    return a;
}

File: Euclid/Euclidean.py
def Euclidean(a,b):
# This function calculates the GCD of two numbers using Euclidean Algorithm
# The algorithm works by repeatedly taking the remainder of the division of two numbers
# until the remainder is zero. The divisor at this point is the GCD.
    temp = 0
    while b:
        a,b = b,a%b
    return a

a = int(input("Enter the first number: "))
b = int(input("Enter the second number: "))
print ("The GCD of",a,"and",b,"is ",Euclidean(a,b))

